tools/logs2seq.py

```diff
@@ -34,73 +34,100 @@
-        logger.info(f"sent : {ctr}")
-        if tosend != b"":
-            requests.post(url, data=tosend)
+class LogSender:
+    def __init__(self, url: str, max_batch_size: int = 100000):
+        self.url = url
+        self.max_batch_size = max_batch_size
+        self.batch = b""
+        self.ctr = 0
+
+
+    def send_log(self, log_message: str):
+        """
+        Send a log message to a seq URL. logs are batched until the maximum batch size is reached (100000b),
+        and then they are sent
+
+        Args:
+            log_message (str): The log message to send
+
+        """
+        if len(self.batch) + len(log_message) > self.max_batch_size:
+            requests.post(self.url, data=self.batch)
+            self.batch = b""
+        self.batch += log_message
+        self.ctr += 1
+        logger.debug(f"sending : {len(self.batch)} bytes")
+
+
+    def read_file(self, file_name: str):
+        """
+        Read the content of a text file, Gzip supported
 
+        Args:
+            file_name (str): The name of the file to read
 
-def process_jsongz_log(url: str, file_name: str):
-    ctr = 0
-    tosend = b""
-    with gzip.open(file_name, "r") as file:
-        for line in file.read().decode("utf-8").split("\n"):
+        Returns:
+            str: The content of the read file
+
+        """
+        if file_name.endswith('.json.tar.gz'):
+            with gzip.open(file_name, 'rt', encoding="utf-8") as file:
+                content = file.read()
+        elif file_name.endswith('.json'):
+            with open(file_name, 'r', encoding="utf-8") as file:
+                content = file.read()
+        return content
+
+
+    def process_json_log(self, file_name: str):
+        """
+        Read a JSON log file, extract log messages, and send them to a Seq server.
+
+        Args:
+            file_name (str): The name of the file to read
+
+        """
+        json_data = self.read_file(file_name)
+        for line in json_data.split('\n'):
             if line.startswith("{"):
                 try:
                     parsed = json.loads(line)
                     if "@t" not in parsed:
                         continue
-                    ctr = ctr + 1
                     log_message = bytes(line + "\n", "utf-8")
-                    if len(tosend) + len(log_message) > 100000:
-                        requests.post(url, data=tosend)
-                        tosend = log_message
-                    else:
-                        tosend += log_message
+                    self.send_log(log_message)
                 except JSONDecodeError as e:
                     logger.warning(f"Failed to parse JSON: {e}")
-        logger.info(f"sent : {ctr}")
-        if tosend != b"":
-            requests.post(url, data=tosend)
-
-if file_name.endswith(".json.tar.gz"):
-    process_jsongz_log(args.url, file_name)
-elif file_name.endswith(".json"):
-    process_json_log(args.url, file_name)
```

Comment:
You've lost that and that is not good !

Reply:
I tried to handle file extensions with the read_file method  for better readability.
Isn't it possible?